### PR TITLE
update md snippets

### PR DESCRIPTION
**Pull request by SimonCropp**

### CODE REVIEW

**Comment by SeanFeldman** [comment]
:white_check_mark: [Build ServiceBus.AttachmentPlugin 0.0.531-preview completed](https://ci.appveyor.com/project/seanfeldman/servicebus-attachmentplugin/builds/26649394) (commit https://github.com/SeanFeldman/ServiceBus.AttachmentPlugin/commit/2e09d84532 by @SimonCropp)

File: src/ServiceBus.AttachmentPlugin.sln.DotSettings [hunk 1]
```diff
@@ -591,6 +591,9 @@ II.2.12 &lt;HandlesEvent /&gt;&#xD;
 	<s:String x:Key="/Default/CodeStyle/Naming/XamlNaming/UserRules/=XAML_005FFIELD/@EntryIndexedValue">&lt;Policy Inspect="True" Prefix="" Suffix="" Style="AaBb" /&gt;</s:String>
 	<s:String x:Key="/Default/CodeStyle/Naming/XamlNaming/UserRules/=XAML_005FRESOURCE/@EntryIndexedValue">&lt;Policy Inspect="True" Prefix="" Suffix="" Style="AaBb" /&gt;</s:String>
 	<s:Boolean x:Key="/Default/Environment/SettingsMigration/IsMigratorApplied/=JetBrains_002EReSharper_002EPsi_002ECSharp_002ECodeStyle_002ECSharpFileLayoutPatternsUpgrade/@EntryIndexedValue">True</s:Boolean>
+	<s:Boolean x:Key="/Default/Environment/SettingsMigration/IsMigratorApplied/=JetBrains_002EReSharper_002EPsi_002ECSharp_002ECodeStyle_002ECSharpKeepExistingMigration/@EntryIndexedValue">True</s:Boolean>
```
Comment (SeanFeldman) [inline src/ServiceBus.AttachmentPlugin.sln.DotSettings]: These should not be here, right?

File: src/ServiceBus.AttachmentPlugin.Tests/DocoUpdater.cs [hunk 2]
```diff
@@ -1,11 +0,0 @@
-﻿using MarkdownSnippets;
-using Xunit;
-
-public class DocoUpdater
-{
-    [Fact]
-    public void Run()
-    {
-        DirectoryMarkdownProcessor.RunForFilePath();
```
Comment (SeanFeldman) [inline src/ServiceBus.AttachmentPlugin.Tests/DocoUpdater.cs]: I'm not sure I like this option. It was very easy to run this test as a part of the test suite.
And I got it working (https://github.com/SimonCropp/MarkdownSnippets/issues/66#issuecomment-520612687).

File: README.md [hunk 3]
```diff
@@ -1,8 +1,10 @@
 <!--
-This file was generate by MarkdownSnippets.
+GENERATED FILE - DO NOT EDIT
+This file was generated by [MarkdownSnippets](https://github.com/SimonCropp/MarkdownSnippets).
 Source File: /README.source.md
-To change this file edit the source file and then re-run the generation using either the dotnet global tool (https://github.com/SimonCropp/MarkdownSnippets#markdownsnippetstool) or using the api (https://github.com/SimonCropp/MarkdownSnippets#running-as-a-unit-test).
+To change this file edit the source file and then run MarkdownSnippets.
```
Comment (SeanFeldman) [inline README.md]: I still like the unit test approach more (see comment below), but wonder what's the benefit of the .NET global tool over unit test.

Comment (SimonCropp) [reply README.md]: >  what's the benefit of the .NET global tool over unit test.

i didnt say there was. however i think there is a benefit of using msbuild over unit tests.

File: src/ServiceBus.AttachmentPlugin.sln.DotSettings (hunk 1 above)
Comment (SimonCropp) [reply src/ServiceBus.AttachmentPlugin.sln.DotSettings]: yeah i will revert that part.

**Comment by SeanFeldman** [comment]
@SimonCropp the `toc` should be after "nuget package" section as that one is breaking TOC.
I have implemented it https://github.com/SeanFeldman/ServiceBus.AttachmentPlugin/commit/0d8ccf09a96cb3447198a685b072e37e11fab9a4 and think will keep the unit test option to generate snippets for now. While the global tool is much more powerful, I prefer simplicity 🙂

**Comment by SeanFeldman** [comment]
Thank you for showing me the `toc` option. That's very nice 🙇

**Comment by SimonCropp** [comment]
just to be clear. this PR does not force u  to use the global tool. it uses the msbuild approach https://github.com/SimonCropp/MarkdownSnippets/blob/master/docs/msbuild.md

**Comment by SeanFeldman** [comment]
Ah, sorry. Ok, then I'll review that as it might be even simpler to regenerate readme file. Perhaps less intuitive (more magic), but simpler. Thamk you again Simon 🙂

**Comment by SimonCropp** [comment]
> Perhaps less intuitive (more magic), but simpler

I am on the fence on that one. i like think of the rendered doco as a build output. so for me it make sense that it to be re-rendered on a build. also means if u break doco (eg missing snippet) your build breaks, which i like.

**Pull request closed without merging**